packages/appinio_swiper/lib/appinio_swiper.dart

```diff
@@ -1,679 +1,743 @@
     this.controller,
-    this.padding = const EdgeInsets.symmetric(horizontal: 20, vertical: 25),
     this.duration = const Duration(milliseconds: 200),
-    this.maxAngle = 30,
+    this.maxAngle = 15,
+    this.invertAngleOnBottomDrag = true,
     this.cardsSpacing = 40,
     this.threshold = 50,
-    this.backgroundCardsCount = 1,
+    this.backgroundCardCount = 1,
+    this.backgroundCardScale = .9,
+    this.backgroundCardOffset,
     this.isDisabled = false,
     this.loop = false,
-    this.swipeOptions = const AppinioSwipeOptions.all(),
-    this.allowUnswipe = true,
-    this.unlimitedUnswipe = false,
+    this.swipeOptions = const SwipeOptions.all(),
     this.onTapDisabled,
     this.onSwipe,
     this.onSwipeCancelled,
-    this.onSwiping,
+    this.onCardPositionChanged,
     this.onEnd,
     this.unswipe,
-    this.direction = AppinioSwiperDirection.right,
-  })  : assert(maxAngle >= 0 && maxAngle <= 360),
+    this.defaultDirection = AxisDirection.right,
+  })  : assert(maxAngle >= 0),
         assert(threshold >= 1 && threshold <= 100),
-        assert(direction != AppinioSwiperDirection.none),
         super(key: key);
 
   @override
   State createState() => _AppinioSwiperState();
 }
 
 class _AppinioSwiperState extends State<AppinioSwiper>
-    with SingleTickerProviderStateMixin {
-  double _left = 0;
-  double _top = 0;
-  double _total = 0;
-  double _angle = 0;
-  double _maxAngle = 0;
-  double _scale = 0.9;
-  double _difference = 40;
-  double _backgroundCardsDifference = 40;
-  final double _backgroundCardsScaleDifference = 0.1;
-  int currentIndex = 0;
-
-  int _swipeType = 0; // 1 = swipe, 2 = unswipe, 3 = goBack
-  bool _tapOnTop = false; //position of starting drag point on card
-
-  late AnimationController _animationController;
-  late Animation<double> _leftAnimation;
-  late Animation<double> _topAnimation;
-  late Animation<double> _scaleAnimation;
-  late Animation<double> _differenceAnimation;
-  late Animation<double> _unSwipeLeftAnimation;
-  late Animation<double> _unSwipeTopAnimation;
-  final Map<int, AppinioSwiperDirection> _swiperMemo =
-      {}; //keep track of the swiped items to unswipe from the same direction
-
-  bool _unSwiped =
-      false; // set this to true when user swipe the card and false when they unswipe to make sure they unswipe only once
-
-  bool _horizontal = false;
-  bool _isUnswiping = false;
-  int _swipedDirectionVertical = 0; //-1 left, 1 right
-  int _swipedDirectionHorizontal = 0; //-1 bottom, 1 top
-
-  AppinioSwiperDirection detectedDirection = AppinioSwiperDirection.none;
+    with TickerProviderStateMixin {
+  static const _defaultBackgroundCardOffset = Offset(0, 40);
 
-  @override
-  void initState() {
-    super.initState();
-    _difference = widget.cardsSpacing;
-    _backgroundCardsDifference = widget.cardsSpacing;
-
-    if (widget.controller != null) {
-      widget.controller!
-        //swipe widget from the outside
-        ..addListener(() {
```

Comment:
This is all handled in `_startActivity` now and there's only a single listener on the controller and a single listener on the animation controller.